lib/addons/core-addon/index.ts

```diff
@@ -17,7 +17,7 @@ export class CoreAddOnProps {
     /**
      * Policy document required by the add-on to allow it to interact with AWS resources
      */
-    readonly policyDocument?: PolicyDocument;
+    readonly policyDocument?: {(partition: string): object};
```

Comment:
Since it is a function let's call it policyDocumentProvider. Also why do we need {} around it? We can drop.

The last point: we can return the PolicyDocument from this function since it is seems the only thing we can do with it.

```
readonly policyDocumentProvider?: (partition: string) => PolicyDocument;
```